### ISSUE TITLE
Rolling cylinder with negative slip

### ISSUE THREAD

**Issue by EelcoHoogendoorn** [google/brax#78]
Im working on adding a torus primitive (with the intent for use in wheeled vehicles).

I am currently working trying to get the contact behaving as it should; its partially working, but I am still working on getting the tangential velocity right, which is still doing funny things.

So I decided to make an env with a single capsule oriented along the y axis as a reference, and giving it a unit impulse along the x axis to get it rolling. It also does funny things. Funny in somewhat different ways; but it seems to settle into a pattern where the rotational velocity exceeds the one required to match the forward velocity at zero slip. I dont think its to do with timesteps and stability; its stable behavior and more substeps dont make a difference.

Ill keep looking and see what I can find, but until then I thought id mention it here. More visual test cases of the physical building blocks would be useful for these kind of things in my opinion.

**Comment by EelcoHoogendoorn** [comment]
Adding a 2x in capsule_plane seems to visually fix this behavior.

`  vel = qp_a.vel + jnp.cross(qp_a.ang, pos - qp_a.pos) * 2`

Is this an interaction with [https://github.com/google/brax/pull/62](url)?

**Comment by EelcoHoogendoorn** [comment]
Of course there shouldnt be a 2x there, assuming 'ang' means 'angular velocity', in the usual sense of the word, where v=cross(omega, r). But appearently thats not what ang means internally, really? The docstring in the QP class gives no indication that this is intentional.

**Comment by erwincoumans** [comment]
So the issue can be closed?
